src/openMVG/matching/regions_matcher.hpp

```diff
@@ -222,14 +223,45 @@ class RegionsMatcherT : public RegionsMatcher
       vec_fDistance.begin(), // distance start
       vec_fDistance.end(),   // distance end
       NNN__, // Number of neighbor in iterator sequence (minimum required 2)
-      vec_nn_ratio_idx, // output (indices that respect the distance Ratio)     
+      vec_valid_matches_idx, // output (indices that respect the distance Ratio)     
       b_squared_metric_ ? Square(f_dist_ratio) : f_dist_ratio);
 
-    vec_putative_matches.reserve(vec_nn_ratio_idx.size());
-    for (size_t k=0; k < vec_nn_ratio_idx.size(); ++k)
+    const size_t n = vec_fDistance.size();
+    
+    // Consider the match valid if the 2 first matches correspond to the "same" feature.
+    //
+    // In [Mishkin 2015] "MODS: Fast and Robust Method for Tow-View Matching" they introduce the FGINN strategy
```

Comment:
Hi,
Yes, @jmelou is experimenting with view synthesis that's why we added that. But it also makes sense for the current state of openMVG as SIFT extract multiple descriptors (with different angles) for one feature.
I'm not sure we need to add an option because if the features are sparse this test will never be true.

Yes, I don't know for the threshold. In the paper, they use a pixel threshold and we kept that. It can make sense because pixel discretization can be considered as a limit in the extraction precision...